presto-spi/src/main/java/com/facebook/presto/spi/function/FunctionNamespaceManagerContext.java

```diff
@@ -14,20 +14,38 @@
 package com.facebook.presto.spi.function;
 
 import com.facebook.presto.common.type.TypeManager;
+import com.facebook.presto.spi.NodeManager;
+
+import java.util.Optional;
 
 import static java.util.Objects.requireNonNull;
 
 public class FunctionNamespaceManagerContext
 {
     private final TypeManager typeManager;
+    private final NodeManager nodeManager;
+    private final Optional<FunctionMetadataManager> functionMetadataManager;
 
-    public FunctionNamespaceManagerContext(TypeManager typeManager)
+    public FunctionNamespaceManagerContext(TypeManager typeManager, NodeManager nodeManager,
+            Optional<FunctionMetadataManager> functionMetadataManager)
     {
         this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.nodeManager = requireNonNull(nodeManager, "nodeManager is null");
+        this.functionMetadataManager = requireNonNull(functionMetadataManager, "functionMetadataManager is null");
```

Comment:
The only creator of this seems to be in `FunctionAndTypeManager`, and it's present there.  Can we make it always present?